### ISSUE TITLE
Donation not updating Credit

### ISSUE THREAD

**Issue by blurryrox** [rathena/FluxCP#94]
I have try searching for solutions around, tried a lot of things but didnt work.
Its seems that the payment is a success but it wont credit on flux.
-My paypal is already a business account.
-IPN url have been set in my paypal
-I am using the latest paymentnotifyrequest.php
-I have tried both live and sandbox.

```
[2017-01-09 01:51:51] Received notification from 173.0.81.33 (notify.paypal.com)
[2017-01-09 01:51:51] Query string: cmd=_notify-validate&payment_type=echeck&payment_date=Mon+Jan+09+2017+01%3A51%3A45+GMT%2B0800+%28Malay+Peninsula+Standard+Time%29&payment_status=Completed&address_status=confirmed&payer_status=verified&first_name=John&last_name=Smith&payer_email=buyer%40paypalsandbox.com&payer_id=TESTBUYERID01&address_name=John+Smith&address_country=United+States&address_country_code=US&address_zip=95131&address_state=CA&address_city=San+Jose&address_street=123+any+street&business=seller%40paypalsandbox.com&receiver_email=seller%40paypalsandbox.com&receiver_id=seller%40paypalsandbox.com&residence_country=US&item_name=something&item_number=AK-1234&quantity=1&shipping=3.04&tax=2.02&mc_currency=USD&mc_fee=0.44&mc_gross=12.34&mc_gross_1=12.34&txn_type=web_accept&txn_id=140816604&notify_version=2.1&custom=xyz123&invoice=abc1234&test_ipn=1&verify_sign=AFcWxV21C7fd0v3bYYYRCpSSRl31AgDFJdhTn1PfaR7sMnx1-Z6E7g4.
[2017-01-09 01:51:51] Establishing connection to PayPal server at www.sandbox.paypal.com:80...
[2017-01-09 01:51:51] Connected. Sending request back to PayPal...
[2017-01-09 01:51:51] Sent 897 bytes of transaction data. Request size: 1049 bytes.
[2017-01-09 01:51:51] Reading back response from PayPal...
[2017-01-09 01:51:51] Notification failed to verify. (recv: )
[2017-01-09 01:51:51] Transaction invalid, aborting.
```

**Comment by Akkarinage** [comment]
Quite  few people have reported IPN failure recently. I'm guessing by the log data that FluxCP is expecting there to be more data if it's sending 897 bytes and getting back 1049 - I thought the two strings were supposed to be identical after the validate command?

@S-anasol - Master Paypal Wizard Sir GM. Have you come across this and/or seen any fixes that would be relevant?

**Comment by S-anasol** [comment]
Maybe paypal doesnt fully support http anymore.

Try replace 80 port with 443?

**Comment by blurryrox** [comment]
Port change even for live? i will try it.

**Comment by S-anasol** [comment]
yeah for both ofc, better anyway 🛡

**Comment by blurryrox** [comment]
well failed using sandbox. I noticed that paypal said that sandbox only support https and live will follow on 30 june 2017. any idea how to change to https to use sanbox?

**Comment by S-anasol** [comment]
so changin 80 to 443 is using https actually.

maybe doesnt work with such old sockopen method :)

**Comment by blurryrox** [comment]
so there's no fix atm?

**Comment by S-anasol** [comment]
fixed

Misconfigured: http://dsro.ru/gyazo/images/5cb99a4fca97d63a2604af40e7e0.png
And doesn't work without ssl

**Comment by S-anasol** [comment]
Added little update for it and also sanbox broken after security fix #95
